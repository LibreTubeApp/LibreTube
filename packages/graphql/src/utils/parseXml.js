import leftPad from 'left-pad';
import xml2js from 'xml2js';

const pad = input => leftPad(input, 2, 0);

export const secondsToHoursAndSeconds = time => {
  const minutes = Math.floor(time / 60);
  const seconds = parseInt(time - minutes * 60, 10);
  const milliseconds = parseInt((((time - minutes * 60) % 1) * 1000), 10);

  if (minutes >= 60) {
    const hours = Math.floor(minutes);
    const minutes = minutes - hours * 60;

    return `${pad(hours)}:${pad(minutes)}:${pad(seconds)}.${milliseconds}`;
  }

  return `${pad(minutes)}:${pad(seconds)}.${milliseconds}`;
};

const convertToVTT = json => (
  `WEBVTT

${json.transcript.text.map(item => {
    const { start, dur }Â = item.$;
    const fromTime = secondsToHoursAndSeconds(start);
    const toTime = secondsToHoursAndSeconds(+start + +dur);
    if (toTime.startsWith('NaN')) {
      console.log('toTime, start, dur', toTime, start, dur);
    }
    return `${fromTime} --> ${toTime}\n${item._}`;
  }).join('\n\n')}`
);

export default xml => new Promise((resolve, reject) => {
  const parseString = xml2js.parseString;
  parseString(xml, function (err, result) {
    if (err) reject(err);
    const formattedResult = convertToVTT(result);
    resolve(formattedResult);
  });
});
