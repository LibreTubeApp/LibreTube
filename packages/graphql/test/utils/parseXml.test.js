import parseXml from '../../src/utils/parseXml';
import mockData from './mockTranscript.js';

describe('the parseXml util', () => {
  it('should parse the XML properly', async () => {
    const parsed = await parseXml(mockData);
    expect(parsed).toMatchSnapshot();
  });
});
