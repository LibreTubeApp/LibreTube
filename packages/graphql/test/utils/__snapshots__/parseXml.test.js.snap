// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the parseXml util should parse the XML properly 1`] = `
"WEBVTT

00:00.20 --> 00:03.439
Hey Vsauce! Michael here. Down here.

00:04.839 --> 00:07.540
But which way is down?

00:07.740 --> 00:14.419
And how much does down weigh? Well, down weighs about a hundredth of a gram per cubic centimeter.

00:14.679 --> 00:20.59
It is light and airy which makes it a great source of insulation and buoyancy for water birds

00:20.239 --> 00:22.379
but if you let go of down...

00:22.940 --> 00:24.839
It falls down.

00:25.39 --> 00:26.579
So that's which way down is.

00:26.679 --> 00:28.960
It's the direction gravity is pulling everything in.

00:29.239 --> 00:32.920
Now, for someone on the other side of the Earth, my down is their up.

00:33.100 --> 00:38.420
But where are falling things going? Why do things fall?

00:38.520 --> 00:43.680
Are they being pushed, or pulled, or is it because of time travel?

00:44.399 --> 00:48.460
First things first: let's turn the Sun into a black hole.

00:49.140 --> 00:55.0
We can do that using universe sandbox 2, this simulator will blow your mind. I love it.

00:55.39 --> 00:59.379
In fact, I love it so much I put a code to get the game for free in the current curiosity box.

00:59.380 --> 01:02.179
If you're not subscribed to the box yet, you are missing out!

01:02.200 --> 01:06.240
Okay, look, for the purposes of this video, we want the solar system.

01:06.239 --> 01:10.419
And here it is. Notice that everything's moving pretty quickly around the Sun.

01:10.439 --> 01:14.719
That's because we currently have the game set so that every second that passes for us,

01:14.719 --> 01:17.219
is 14 days, almost, in the game.

01:17.219 --> 01:20.739
If I change this to one second,

01:20.739 --> 01:23.280
We're looking at the solar system in real time.

01:23.579 --> 01:25.879
You'll notice that it almost looks like it's frozen.

01:26.420 --> 01:31.599
even though the earth is traveling around the Sun at about 30 km/s, it barely appears to be moving.

01:31.760 --> 01:35.640
That is how vast space is. Anyway, let's go back to 14 days

01:35.659 --> 01:37.560
I like that motion. Now look at the Sun

01:37.569 --> 01:43.258
It is not, currently, a black hole, but we can change that. What we need to do is compress the Sun.

01:43.390 --> 01:48.719
So let's lock its mass so that it doesn't change while we make its radius smaller.

01:49.299 --> 01:51.959
Let's make its radius as small as we can.

01:52.579 --> 01:58.859
And, oh, where'd it go? Well it's still there, it's just become a black hole.

01:59.140 --> 02:03.439
Pretty spooky, but now, let's look at the rest of the solar system.

02:03.540 --> 02:06.819
Alright, zooming out and-

02:06.819 --> 02:07.959
huh.

02:07.959 --> 02:11.120
Nothing's... changed. I mean something's changed.

02:11.240 --> 02:16.0
It's colder and darker, but nothing's flying off into space or getting sucked in.

02:16.259 --> 02:20.340
You see,  by shrinking the sun, we didn't change the direction of down for the planets.

02:20.340 --> 02:25.739
They're always being pulled by gravity towards its middle and making it smaller didn't move where the middle was.

02:25.919 --> 02:31.799
But also, the strength of that force pulling them to the middle of the sun stayed the same.

02:32.259 --> 02:35.419
That gives us a clue as to what down is.

02:35.620 --> 02:39.980
The clue is the other thing we didn't change: mass.

02:40.259 --> 02:45.199
Mass is a measure of how hard it is to accelerate something; to change its motion.

02:45.460 --> 02:49.400
Now right now, these two balls have zero motion relative to me.

02:50.259 --> 02:52.560
Slapping around this hollow plastic ball

02:54.340 --> 02:58.109
is pretty easy, but doing the same to this solid steel ball

03:01.419 --> 03:02.679
is a lot harder.

03:03.120 --> 03:05.939
Now gravity and weight have nothing to do with this.

03:06.479 --> 03:10.280
Gravity acts downward, not against my horizontal slapping.

03:10.280 --> 03:16.400
Of course, gravity does contribute to friction, but friction works against me when I start moving the ball,

03:16.598 --> 03:19.18
but works with me when I stop the ball.

03:19.19 --> 03:22.739
And the steel ball is harder to stop than the plastic ball.

03:23.79 --> 03:27.149
The difference is mass. The steel ball is more massive

03:27.149 --> 03:30.939
It's more resistant to having its motion changed.

03:30.939 --> 03:36.699
Mass is an intrinsic property; it does not depend on what's around or change from place to place.

03:36.699 --> 03:40.619
It can sometimes be thought of as the amount of matter something has.

03:40.620 --> 03:43.960
Your mass is the same regardless of where you are.

03:43.960 --> 03:47.760
On the moon, on earth, in the middle of intergalactic space floating around.

03:48.0 --> 03:53.300
But all of this said, mass does seem to care about what's around.

03:53.819 --> 03:56.359
Mass loves company.

03:57.159 --> 04:03.359
Things with mass and/or energy are attracted together by a force that we call gravity.

04:03.609 --> 04:08.189
The feeling of gravity is just you and the earth being attracted to one another.

04:08.639 --> 04:13.119
Now every portion of an object with mass attracts other portions towards it.

04:13.479 --> 04:18.860
The average of all this pulling is an attraction between centers of mass.

04:19.19 --> 04:26.500
Giant things like Earth exert an obvious pull, but everything does. Even a baseball.

04:26.579 --> 04:32.219
These two baseballs are attracted together by their own gravities.

04:36.160 --> 04:42.920
Except their masses are so small, the force is minuscule, and it can't overcome friction or push air out of the way.

04:43.240 --> 04:44.889
They're never gonna come together

04:44.889 --> 04:51.809
But if you put two baseballs one meter apart in the middle of empty space where no other forces could act on them

04:52.29 --> 04:56.578
They would literally fall together and collide.

04:57.129 --> 05:00.538
It would take three days to happen, but it would.

05:01.598 --> 05:07.438
Isaac Newton found that the strength of the force bringing two things together is equal to the product of their masses

05:08.319 --> 05:15.389
Divided by the distance between their centers of mass squared times big G, the gravitational constant.

05:15.788 --> 05:20.788
If you make one of two objects more massive, or move them closer together, the force will be

05:21.9 --> 05:25.379
Stronger and this force of attraction is what we call weight

05:26.50 --> 05:30.270
So mass is intrinsic whereas weight depends on what's around

05:31.459 --> 05:35.138
Now a weird thing happens when you weigh yourself on most scales

05:35.420 --> 05:39.520
Weight is a force but scales display pounds or kilograms

05:39.649 --> 05:42.938
Which are units of mass what's going on?

05:42.939 --> 05:44.839
Is that a scale is

05:44.839 --> 05:46.699
activated by a force?

05:46.699 --> 05:47.288
Any force.

05:47.290 --> 05:51.999
It doesn't have to be caused by gravity. The scale then displays what amount of mass

05:52.250 --> 05:57.310
Near the surface of the earth would be attracted to the earth with the force. It's detecting

05:57.949 --> 06:03.818
Now since scales tend to be used on the surface of the earth by people on which the only force acting is gravity

06:04.40 --> 06:11.468
They tend to not be very far off, but they can be easily tricked and they further lead to the confusion between mass and weight

06:12.949 --> 06:20.288
Notice that weight is mutual. You are pulled down by earth with the same force that you pull up on earth.

06:20.839 --> 06:23.258
According to a scale I weigh

06:23.990 --> 06:25.990
180 pounds on earth

06:30.759 --> 06:35.430
And the earth weighs 180 pounds on me

06:36.699 --> 06:40.889
but because the Earth's mass is so much greater than my own and

06:41.560 --> 06:46.530
Because the more massive something is the more it resists being moved our

06:46.810 --> 06:50.730
Equal and opposite weight forces accelerate me a lot more than the earth

06:51.160 --> 06:58.588
If you drop a pencil from a height of 6 feet the pencil doesn't just fall to the earth more precisely

06:58.870 --> 07:00.870
They both come together.

07:01.269 --> 07:03.718
They're attracted to each other by equal forces

07:03.720 --> 07:06.29
but the same force moves the pencil a

07:06.250 --> 07:11.369
Lot more than the earth when you let go of the pencil the earth is literally pulled up

07:11.439 --> 07:16.889
To the pencil by the pencils own gravity a distance equal to about 9 trillion

07:17.470 --> 07:19.650
the width of a proton. That

07:20.170 --> 07:25.110
same force moves the pencil the remaining distance, which is still pretty much six feet

07:26.589 --> 07:32.219
At the height of the International Space Station's orbit you and earth are attracted about

07:33.69 --> 07:39.149
10% less than when you're on the surface about eight point eight times your mass but not zero

07:39.819 --> 07:46.678
for this reason weightless astronauts in zero gravity are neither weightless nor in zero gravity

07:47.470 --> 07:51.720
their weight force fails to bring them and earth together because they move horizontally

07:51.970 --> 07:57.690
So quickly that they fall. Just as fast as Earth's surface curves away from them and

07:58.300 --> 08:02.759
Even though they're experiencing 90 percent of the gravity you and I are feeling right now

08:03.189 --> 08:04.750
That's why they don't just fly away

08:04.750 --> 08:11.879
There are no forces called g-forces to resist their weight since everything around them is falling too

08:12.310 --> 08:16.170
It's resistance to your weight force stress

08:17.79 --> 08:23.188
Deformation that is needed for you to feel weight what astronauts in orbit actually lack is

08:24.129 --> 08:26.129
apparent weight

08:26.350 --> 08:29.460
likewise a helium balloon has weight

08:29.459 --> 08:35.298
I mean, it's made out of matter it clearly has mass so it's attracted to the earth

08:36.28 --> 08:38.119
Let's try to measure its weight force

08:41.9 --> 08:43.9
That's eight

08:43.979 --> 08:46.759
Okay it has negative apparent weight

08:46.889 --> 08:52.580
That's because its attraction to the earth is weaker than the buoyant forces from the air around it that push it up

08:52.860 --> 08:54.500
Now while it moves up

08:54.500 --> 09:01.700
It pushes air molecules down, but they transfer that force widely. Not just directly down onto the scale

09:02.278 --> 09:07.369
Buoyant forces are caused by the fact that whenever you are immersed in a fluid like water or air

09:08.190 --> 09:10.760
Molecules lower down are at greater pressure

09:10.980 --> 09:13.310
That are being pressed by the weight of all the molecules

09:13.440 --> 09:18.469
Above them and are closer to earth so they're pulled to it with a stronger force now having greater pressure

09:18.570 --> 09:21.709
Means they pack a bigger punch when they collide with things.

09:22.259 --> 09:23.458
So

09:23.458 --> 09:25.548
horizontally those collisions cancel out

09:25.620 --> 09:32.239
But vertically the stronger collisions from below went out providing lift a buoyant force

09:32.549 --> 09:38.478
This even happens on your own body across its surface area air lifts you with the force of about one

09:38.610 --> 09:45.589
Newton which is equal to the weight force of an apple so if you weighed yourself in a vacuum you would weigh about

09:45.839 --> 09:47.839
this much more

09:48.269 --> 09:54.678
But that's not all Earth's spin causes it to bulge at the equator so the closer you are to it

09:54.679 --> 10:01.759
The further you are from Earth's center of mass and the less your actual gravitational weight will be down is

10:02.100 --> 10:04.190
Always changing, I mean

10:04.289 --> 10:11.28
where is Earth's center of mass? It would always be the same as Earth's geometric middle if Earth's

10:11.250 --> 10:18.379
composition was uniform, but earth contains pockets of massive rock at different depths water mountains

10:18.379 --> 10:23.778
It's got moving changing insides and air and seasonal ice and though they're far away

10:24.240 --> 10:30.798
Gravity extends forever from everything so the moon the Sun the planets all of them pull on you

10:31.708 --> 10:32.819
negligibly,

10:32.820 --> 10:39.289
But truly. You weigh about a millionth of your weight less when the moon is directly above you

10:39.659 --> 10:46.740
This chunky shifting balance of material on earth and ever where else in the universe means that down is

10:47.200 --> 10:48.250
always

10:48.250 --> 10:50.849
changing on top of that Earth's spin

10:51.100 --> 10:56.820
Skews what you consider the direction of down away from its center of mass because the push you get from Earth's spin

10:57.39 --> 11:01.439
Seems to slightly lift you reducing your apparent weight and bending down

11:01.929 --> 11:03.309
towards the equator

11:03.309 --> 11:10.500
The net result is an apparent weight reduction at the equator of about half of a percent if a scale guesses your mass must be

11:10.929 --> 11:14.38
200 pounds at the poles it'll guess that you're

11:14.769 --> 11:18.239
199 at the equator. The 9.8

11:18.309 --> 11:24.689
Multiplier used so often in physics is calculated based on how these factors affect someone at 45 degrees latitude

11:25.208 --> 11:30.418
all of these influences on the direction of down result in a total vertical deflection.

11:30.820 --> 11:34.769
That's only ever at most a few arc seconds anywhere on earth

11:35.49 --> 11:41.819
That's not enough to be felt, but changes in direction and strength can be used to study the shape of the seafloor

11:42.429 --> 11:47.370
Determine what's under you or even help you discover ancient buried rooms?

11:48.399 --> 11:53.969
Point is all of our downs aren't a bunch of radially symmetric lines

11:54.578 --> 11:57.28
Down is an uncombed mess.

11:57.549 --> 12:03.240
Now since solids don't flow they can have shapes that don't pay much mind to this but water can

12:03.639 --> 12:11.249
Flow so ignoring influences like wind and tides the surface of oceans and lakes and puddles is always perpendicular

12:11.379 --> 12:17.70
To down if water could pass through land, or if earth were submerged in water

12:17.860 --> 12:21.418
Gravity would be the same everywhere along its bumpy surface

12:21.789 --> 12:26.250
Such a surface is called a geoid and can be drawn at any altitude

12:26.708 --> 12:32.458
If you wanted to build a table that completely enclosed the earth it would have to have rolling undulation

12:32.458 --> 12:35.338
Z' nearly 100 meters at some points in order to be

12:35.889 --> 12:39.629
Level so that a ball placed anywhere on it wouldn't roll

12:40.240 --> 12:43.620
Here is Earth's G. I exaggerated a thousand times

12:44.110 --> 12:48.180
You'd weigh about a hundredth of a percent less a few grams here?

12:48.179 --> 12:53.908
Then you would say here where gravity is a bit stronger point is the strength and direction

12:54.259 --> 12:58.728
Down is variable by location and changes over time

12:59.489 --> 13:04.278
So down is a fluctuating vector easy enough except?

13:04.739 --> 13:08.538
Why should matter attract matter in the first place?

13:09.539 --> 13:13.759
Isaac Newton was able to describe attraction, but not explain it

13:14.369 --> 13:19.369
humanity got closer however when Albert Einstein introduced his general theory of relativity

13:20.639 --> 13:23.988
Einstein thought a lot about the fact that everything falls to the ground

13:24.328 --> 13:26.507
At the same rate no matter

13:26.509 --> 13:33.108
How massive something is when dropped it will accelerate towards the earth down gaining about 9.8

13:33.109 --> 13:37.249
Meters per second for every second that it falls. That means that a hammer

13:37.589 --> 13:44.628
That's quite massive, and a not so massive feather when dropped from the same height will hit the ground at the same time

13:47.818 --> 13:49.818
Okay, what just happened...

13:50.278 --> 13:51.989
was an \\"air\\"ror.

13:51.989 --> 13:53.38
awkward laugh

13:53.38 --> 13:56.389
In order to fall through air a thing has to push air out of the way

13:56.548 --> 14:01.759
But if it has a large surface area and a low weight force it will have a lot of air to move

14:01.889 --> 14:07.278
But won't be able to move that air very quickly. In a vacuum, things do fall at the same rate regardless

14:07.460 --> 14:12.940
Of mass. This was famously demonstrated by Apollo 15 commander David Scott on the moon

14:14.39 --> 14:18.659
And I'll, uh, drop the two of them here and hopefully, they'll hit the ground at the same time.

14:20.279 --> 14:21.259
How 'bout that?

14:21.259 --> 14:26.340
That's weird right? I mean if a more massive object is pulled with a greater force

14:27.149 --> 14:30.589
Shouldn't it fall faster? Well Newton's explanation was simple:

14:31.409 --> 14:34.159
Larger masses are attracted with greater forces

14:34.589 --> 14:39.438
But will also require more force to be accelerated the same as a less massive thing

14:39.899 --> 14:46.8
Something a hundred times more massive might require a hundred times the force, but it will be pulled by gravity

14:46.769 --> 14:51.769
100 times more so everything falls to earth at the same rate

14:52.499 --> 14:54.499
What a fun coincidence right?

14:56.970 --> 14:58.739
Maybe not

14:58.739 --> 15:02.749
Einstein realized that there's another way for things to appear to fall together

15:03.889 --> 15:05.399
of their masses

15:05.399 --> 15:12.888
Imagine a feather and a hammer floating in space in a room if the room is suddenly accelerated up at 9.8

15:13.318 --> 15:18.78
M/s^2 the feather and the hammer will hit the floor at the same time

15:18.720 --> 15:24.319
furthermore whether it's the room coming up to meet them or gravity being suddenly switched on

15:24.659 --> 15:27.918
Neither object will feel any force pushing them

15:28.78 --> 15:33.317
There's no way to tell which of these happen. This is Einstein's famous

15:34.19 --> 15:35.399
equivalence principle

15:35.399 --> 15:39.618
He once admitted that his greatest thought ever was that of a man

15:40.438 --> 15:42.438
falling off of a roof

15:42.629 --> 15:48.557
While falling the man would not feel any forces on him even though. He's speeding up

15:49.528 --> 15:50.909
freefall is

15:50.909 --> 15:57.289
Indistinguishable from floating alone in space from having no forces on you from not being moved

15:57.688 --> 16:03.557
What if gravity isn't a force at all what if things fall not because they're being pushed or pulled

16:03.808 --> 16:07.788
But because they're not being pushed or pull

16:08.100 --> 16:12.619
To see how this could be we need to talk about straight lines

16:13.649 --> 16:17.388
What I have here is a retractable ID badge holder

16:17.489 --> 16:20.209
This is a great way to test for straight paths

16:20.698 --> 16:24.168
Because the string is always kept taut the card

16:24.168 --> 16:26.749
I have behind has two lines drawn on it

16:26.749 --> 16:32.57
And if while I pull the string out it always stays between those two lines

16:32.58 --> 16:34.458
I will know that I never turned

16:34.860 --> 16:40.699
While I pulled it because any turn will translate into a different angle between the lines on the card and the string

16:40.948 --> 16:47.177
Now if I put two of these on a flat table and pull them out, always ensuring that they go straight ahead

16:47.399 --> 16:55.128
They will never meet. They will be forever parallel, but now let's put them on a sphere a curved surface

16:55.798 --> 17:02.119
Again, I pull both strings forward making sure that they always are pulled out straight. No turning

17:03.58 --> 17:05.58
Wait they came together

17:06.130 --> 17:08.309
Well, they didn't turn look

17:08.710 --> 17:15.840
Maybe there's some kind of weird force that pulled my hands together and just like gravity. I didn't feel it, but it happened

17:16.390 --> 17:17.560
No

17:17.559 --> 17:22.440
What happened was not the result of a force it was just a natural result of?

17:23.140 --> 17:29.970
Curvature you might be thinking wait a second are those really straight lines. I mean they don't look that straight to me

17:30.640 --> 17:37.500
Also, what if they've just moved along latitude lines then they've never come together and those look pretty darn straight

17:38.440 --> 17:45.509
But they're not a straight line never turns and although latitude lines look straight at first glance

17:46.269 --> 17:48.269
following one requires

17:48.279 --> 17:54.148
Turning to find straight line paths on surfaces whether they're flat like this or curved

17:54.329 --> 17:58.318
I love the written text now you can use an actual ribbon

17:58.319 --> 18:04.168
But I have found that a strip of paper works. Even better. Let's take a look at this path right here

18:04.480 --> 18:10.318
It's straight at first, but then it curves now if two people are traveling along this curve

18:10.319 --> 18:16.79
And they want to stay together the person on the inside will have to cover a shorter distance than the person on the outside

18:16.990 --> 18:20.429
Since both sides of this strip of paper cannot change their lengths

18:20.430 --> 18:25.409
They'll help us find a straight path if the strip of paper can lay flat

18:25.599 --> 18:28.888
We'll know that we have found a straight line and as you can see

18:29.230 --> 18:32.788
The strip can lay flat and follow the straight part of this path

18:32.890 --> 18:36.150
But when it comes to the curve in order to follow the path now

18:36.339 --> 18:40.168
The strip well it has too much material on the inside and that material

18:40.329 --> 18:46.408
Bunches up and leaves the plane therefore we know that this part of the path is not straight

18:47.769 --> 18:52.199
Let's use the ribbon test to find straight lines on the surface of a cone

18:52.660 --> 18:58.889
Well from the looks of it aligned directly from the base to the tip seems like it would be straight and sure enough

18:58.930 --> 19:04.199
Yeah, the ribbon lays flat on that path, but what about a ring around the cone?

19:05.79 --> 19:11.460
Nope doesn't work shorter distances around nearer the tip of the cone mean that there's too much ribbon up at the top

19:11.460 --> 19:13.180
So doesn't lay flat

19:13.180 --> 19:19.469
Let's see what else is there though besides this well if I start here and just allow the ribbon

19:20.920 --> 19:22.690
to lay flat

19:22.690 --> 19:26.370
Huh I get a little curvy looking shape like this

19:26.429 --> 19:32.759
I say curvy looking because while to someone say at the base this path might seem to go up

19:33.190 --> 19:39.779
Slow down change direction and then fall down faster and faster since a ribbon on such a path is flat

19:39.940 --> 19:43.529
It's actually for inhabitants on the cones surface

19:44.109 --> 19:48.58
Perfectly straight if we trace the ribbons path on to the cone

19:48.58 --> 19:55.138
We can see this clearly because a cone can be flattened a straight line on a curved surface is called a geodesic

19:56.529 --> 20:00.928
Here is a geodesic on a sphere the Equator is one

20:01.509 --> 20:05.368
Here's another a line of latitude is not a geodesic

20:05.369 --> 20:09.478
It's not a straight line to see why let's try to follow it with the ribbon you

20:10.89 --> 20:13.139
Know what I have to keep kind of lifting it

20:13.329 --> 20:17.428
Yeah, see distances around the sphere becomes shorter as we go up

20:17.500 --> 20:19.799
So there's too much material on the ribbon up here

20:19.799 --> 20:24.628
and it leaves the surface this path contains turns and in order to turn a

20:24.759 --> 20:27.749
Force has to act on you if no forces

20:27.750 --> 20:34.618
Did this is the path you would take notice that the ribbon begins moving due east, but then falls

20:35.349 --> 20:37.349
south

20:37.390 --> 20:39.390
Falls

20:39.940 --> 20:44.549
Einstein realized that curvature could cause things to be seemingly attracted to one another

20:45.69 --> 20:48.720
Without needing to invent the existence of forces like gravity

20:49.89 --> 20:56.188
but attraction only happens if things move along the surface if they stay still they

20:56.589 --> 21:02.369
Well, they don't come together so for something at rest. How does falling begin?

21:02.369 --> 21:07.919
I mean the thing has to move in this direction, but it's at rest right well

21:07.990 --> 21:08.490
Yes

21:08.490 --> 21:14.189
But it's only at rest in space and that's not the whole story up down

21:14.650 --> 21:22.469
Forward backward and left-right are all you need to describe where an event occur but a complete description will also need to describe

21:23.319 --> 21:28.558
When together these four dimensions form the setting in which everything in our universe happens

21:30.279 --> 21:31.660
Space-time

21:31.660 --> 21:38.790
Since we can talk about a falling pencil using just one spatial dimension up and down we can use a piece of paper to model

21:39.9 --> 21:43.108
Space-time for it. Okay, so we've got up and down, but we have to add another

21:43.660 --> 21:45.460
direction the pencil moves in

21:45.460 --> 21:46.299
time

21:46.299 --> 21:50.338
Now if no forces act on the pencil it won't move through space

21:50.339 --> 21:54.839
It will only get older and as you can see if all it does is get older

21:54.839 --> 22:00.29
It won't fall if space-time was flat when I let go of the pencil

22:00.640 --> 22:07.650
it wouldn't go anywhere, but now let's allow the earth which is massive to manipulate space-time into say a

22:08.170 --> 22:09.190
cone

22:09.190 --> 22:13.980
Now with no forces acting on it every part of the pencil follows a straight line

22:14.79 --> 22:20.609
But on a cone as we saw earlier such a path will look like this it will fall

22:20.859 --> 22:24.899
This is because distances around the cone are shorter higher up

22:25.210 --> 22:29.429
Time runs faster further from a massive object, but to go straight

22:29.500 --> 22:34.78
Not turn every part of the pencil must cover an equal distance in space-time

22:34.630 --> 22:36.190
like this

22:36.190 --> 22:43.289
Only when the pencil hits the earth does the repulsion of their mutual electrons provide a force pushing the pencil off a geodesic

22:43.750 --> 22:49.949
For the earth time is a series of slices from this evolution the pencils force-free

22:50.259 --> 22:57.868
Geodesic is why it falls not a push or pull just the pencils natural tendency to follow a straight line

22:58.148 --> 23:04.348
Until something acts on it now. We only used one dimension of space and one of time because

23:04.869 --> 23:11.638
Visualizing our universes three of space and one of time would take us beyond the limits of what could be shown on paper or screens

23:12.99 --> 23:13.388
but math

23:13.388 --> 23:15.158
Can take us there

23:15.159 --> 23:18.809
General relativity allows us to calculate how much mass and energy

23:19.148 --> 23:26.398
Curved space-time and has been used to explain things that Newton's older theory of falling as the result of

23:26.648 --> 23:34.377
Forces couldn't like anomalies in the orbit of mercury which orbits nearest the Sun and is therefore most affected by the sun's grip on

23:34.839 --> 23:41.308
space-time many other experiments have confirmed general relativity's picture of the universe fitting the conclusion that

23:42.9 --> 23:44.9
There is no

23:44.619 --> 23:46.619
Gravity there's just

23:47.79 --> 23:52.399
Space-time its curvature and a us in it. As John Wheeler famously put it

23:52.960 --> 23:59.759
Space-time grips mass telling it how to move mass grips space-time telling it how to curve?

24:00.669 --> 24:07.439
Relative to the earth we don't move very fast even jet airplanes move negligibly close to the speed of light

24:07.599 --> 24:10.588
So relative to earth we move almost

24:10.960 --> 24:17.548
Exclusively through time as such we are more affected by the way time is curved by mass

24:17.740 --> 24:19.659
Than how space is curved?

24:19.659 --> 24:24.869
This has led many to claim that for the most part you feel as though you're being pushed into the ground

24:25.240 --> 24:27.419
Not because of a force called gravity

24:27.519 --> 24:33.449
But because time is moving faster for your head than for your feet down is

24:33.759 --> 24:41.249
Relative and always changing, but it exists because of and is always in the direction of slower time

24:41.980 --> 24:45.838
Bertrand Russell called this the law of cosmic laziness

24:46.690 --> 24:53.759
Everything is naturally steered towards where time is slowest we call this falling

24:54.429 --> 24:56.429
going down

24:56.529 --> 24:59.848
So you don't have to keep anything on the down-low

25:00.579 --> 25:03.688
time will take care of that for you and

25:04.750 --> 25:06.490
and as always

25:06.490 --> 25:08.490
Thanks for watching

25:19.190 --> 25:24.370
Remember that you can support Vsauce and Alzheimer's research by subscribing to the Vsauce

25:24.920 --> 25:29.320
curiosity box the current one comes with a code to get a free copy of universe sandbox 2

25:29.390 --> 25:35.590
Which is amazing and a whole host of other science toys and tools picked by myself Jake and Kevin

25:35.589 --> 25:41.980
I love it all so I hope to see you at brain candy live. We are coming to many many cities very soon

25:42.130 --> 25:45.340
Hopefully one near you by going to the show you can see Adam

25:45.339 --> 25:51.220
And I doing things that will you may not have seen us do before we also explore the science and common

25:51.769 --> 25:54.490
misconceptions behind all things

25:55.789 --> 26:03.158
Err. Maybe have said too much. Maybe not I hope to see you there and as always thanks for watching"
`;
